core/src/test/scala/com/softwaremill/sttp/testing/SttpBackendStubTests.scala

```diff
@@ -155,6 +157,25 @@ class SttpBackendStubTests extends FlatSpec with Matchers with ScalaFutures {
 
   }
 
+  it should "not hold the calling thread when passed a future monad" in {
```

Comment:
I'm not sure this is a good test ... the fact that the calling thread isn't held is ensured by the way the `Future` is constructed - not by anything in sttp itself. For this test to fail sttp would need to block on futures somewhere, which would be very hard to do given the generic `Monad` implementation :)

Maybe a better test case here would be that when using `.thenRespondWithMonad`, the specified response is indeed returned, when the future returned by `.send` completes?

Reply:
That was my first attempt. Getting `Future` from `.send` and checking if that was the same as passed to `.thenRespondWithMonad`. However, it was not the same because of mapping in `SttpBackendStub.tryAdjustResponseType` (line 171).

Unless you mean just testing basic functionality of `.thenRespondWithMonad`? It's true that it's hard to block on future in a case we have here. It's all because that was our test scenario, which STTP was not feasible for, thus the change :)

Reply:
I meant testing the basic functinality of `.thenRespondWithMonad`. I don't think it matters if it's the exact same future or not :)

Reply:
Test for response retrieved from Monad added